system_tests_async/noxfile.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2016 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Noxfile for automating system tests.
+
+This file handles setting up environments needed by the system tests. This
+separates the tests from their environment configuration.
+
+See the `nox docs`_ for details on how this file works:
+
+.. _nox docs: http://nox.readthedocs.io/en/latest/
+"""
+
+import os
+import subprocess
+
+from nox.command import which
+import nox
+import py.path
+
+
+HERE = os.path.abspath(os.path.dirname(__file__))
```

Comment:
I'm not familiar enough with nox to know, but, assuming this is mostly copied from system_tests/noxfile, is it possible to re-factor and deduplicate some of this?

Reply:
We shouldn't have two noxfiles if at all possible

Reply:
Resolved this to have 1 noxfile for both the sync and async system tests - essentially split up the python 2/3 dependencies and run both separately (similar to the sync approach)